centaur/src/main/scala/centaur/test/metadata/WorkflowFlatMetadata.scala

```diff
@@ -85,10 +92,69 @@ case class WorkflowFlatMetadata(value: Map[String, JsValue]) extends AnyVal {
 
 object WorkflowFlatMetadata {
 
-  def fromConfig(config: Config): ErrorOr[WorkflowFlatMetadata] =
-    config.extract[Map[String, Option[String]]] match {
-      case Result.Success(m) => Valid(WorkflowFlatMetadata(m safeMapValues { _.map(JsString.apply).getOrElse(JsNull) }))
-      case Result.Failure(_) => invalidNel(s"Metadata block can not be converted to a Map: $config")
+  def fromConfig(config: Config): ErrorOr[WorkflowFlatMetadata] = {
+    val metadataValues = config.entrySet().toArray(Array[util.Map.Entry[String, ConfigValue]]()).map { value =>
+      val v: ConfigValue = value.getValue
+      val key: String = value.getKey.stripPrefix("\"").stripSuffix("\"")
+      v.valueType() match {
+        case ConfigValueType.BOOLEAN => key -> JsBoolean.apply(v.unwrapped().asInstanceOf[Boolean])
+        case ConfigValueType.NUMBER =>
+          val num = v.unwrapped().asInstanceOf[Number]
+          if (num.intValue() == num.doubleValue()) {
+            key -> JsNumber.apply(num.intValue())
+          } else {
+            key -> JsNumber.apply(num.doubleValue())
+          }
+        case ConfigValueType.LIST =>
+          key ->
+            parseArray(
+              v.unwrapped()
+                .asInstanceOf[util.List[Object]]
+                .asScala
+                .toVector
+            )
+        case ConfigValueType.NULL => key -> JsNull
+        case _ =>
+          /*
+            FIXME/TODO:
+
+            As part of WX-1629
+            https://broadworkbench.atlassian.net/jira/software/c/projects/WX/boards/174?selectedIssue=WX-1629, this
```

Comment:
I don't think we need to put a URL here at all, but if we want to, it should be `https://broadworkbench.atlassian.net/browse/WX-1629`